default/scripting/techs/ship_weapons/ship_weapons.py

```diff
@@ -12,15 +13,30 @@ def WEAPON_BASE_EFFECTS(part_name: str):
     NamedReal(name=part_name + "_PART_CAPACITY", value=PartCapacity(name=part_name))
     NamedReal(name=part_name + "_PART_SECONDARY_STAT", value=PartSecondaryStat(name=part_name))
 
-    # The following is really only needed on the first resupplied turn after an upgrade is researched, since the resupply currently
-    # takes place in a portion of the turn before meters are updated, but currently there is no good way to restrict this to
-    # only that first resupply (and it is simply mildly inefficient to repeat the topup later).
+    # side-note: set_max_meters_effects_group is not necessary for MaxCapacity iff WEAPON_UPGRADE_CAPACITY_EFFECTS is applied
+    set_max_meters_effects_group = EffectsGroup(
+        scope=EMPIRE_OWNED_SHIP_WITH_PART(part_name),
+        accountinglabel=part_name,
+        priority=DEFAULT_PRIORITY,
+        effects=[
+            SetMaxCapacity(partname=part_name, value=PartCapacity(name=part_name)),
+            SetMaxSecondaryStat(partname=part_name, value=PartSecondaryStat(name=part_name)),
+        ],
+    )
+
+    # The topup_effects_group is necessary for
+    # 1) first resupplied turn after an upgrade is researched, since the resupply currentlytakes place in a portion of the turn before meters are updated
+    # 2) any turn with an arbitrary "temporary" change of Max values (e.g. PLC_FLANKING); if we had only tech effects we could restrict this
```

Comment:
what does it mean to have "only tech effects"?

Reply:
i am not sure if the info is meaningful/how to phrase. probably will simply remove the second part